### PR TITLE
Replace "Always_show_goals" with "Toggle_showing_goals"

### PR DESCRIPTION
**Pull request by naomimyselfandi**
The "Always_show_goals" flag is a no-op on anything other than a training mission. This replaces it with a new flag which toggles whether the goals are shown instead, in keeping with the existing "Toggle_debriefing" flag.

### CODE REVIEW

**Review by Goober5000 — CHANGES_REQUESTED** [review]
Everything here looks good.  But a few more changes will need to be made in fred2\missionnotesdlg.cpp and qtfred\src\ui\dialogs\MissionSpecDialog.cpp due to the variable change.

**Comment by naomimyselfandi** [comment]
Huh. I thought I'd also built FRED, but I guess I mustn't have.

**Review by Goober5000 — CHANGES_REQUESTED** [review]
Two more things.  Happily, these only need to be done in FRED, as whoever added the qtFRED flags thought ahead.

File: fred2/missionnotesdlg.cpp [hunk 1]
```diff
@@ -399,7 +399,7 @@ BOOL CMissionNotesDlg::OnInitDialog()
 	m_autpilot_cinematics = (The_mission.flags[Mission::Mission_Flags::Use_ap_cinematics]) ? 1 : 0;
 	m_2d_mission = (The_mission.flags[Mission::Mission_Flags::Mission_2d]) ? 1 : 0;
 	m_no_autpilot = (The_mission.flags[Mission::Mission_Flags::Deactivate_ap]) ? 1 : 0;
-	m_always_show_goals = (The_mission.flags[Mission::Mission_Flags::Always_show_goals]) ? 1 : 0;
```
[the remark this answers is not in the document]
Comment (naomimyselfandi) [reply fred2/missionnotesdlg.cpp:402]: I think you're commenting on the removed version of this line.

File: fred2/missionnotesdlg.cpp [hunk 2]
```diff
@@ -289,7 +289,7 @@ void CMissionNotesDlg::OnOK()
     The_mission.flags.set(Mission::Mission_Flags::Deactivate_ap, m_no_autpilot != 0);
 	
 	// always show mission goals
```
[the remark this answers is not in the document]
Comment (naomimyselfandi) [reply fred2/missionnotesdlg.cpp]: Good catch.

File: fred2/missionnotesdlg.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (Goober5000) [reply fred2/missionnotesdlg.cpp:402]: I'm referring to the `m_always_show_goals` member variable of CMissionNotesDlg, not the flag name.

[the remark this answers is not in the document]
Comment (naomimyselfandi) [reply fred2/missionnotesdlg.cpp:402]: Wow, I totally didn't see that. I caught a few other stale names and I think I've updated all of them.

**Review by Goober5000 — APPROVED** [review]
Looks good now. 👍 

I would also suggest removing `IDC_TOGGLE_SHOWING_GOALS2`, since that isn't used anywhere, but that doesn't need to hold up the PR.

File: code/missionui/missionbrief.cpp [hunk 3]
```diff
@@ -885,10 +885,12 @@ void brief_init()
 		return;
 	}
 
-	if (The_mission.flags[Mission::Mission_Flags::Always_show_goals] || !(The_mission.game_type & MISSION_TYPE_TRAINING))
+	// Show the goals slide iff we're in a training mission with the "toggle goals" flag, or a normal mission without it.
+	if (The_mission.flags[Mission::Mission_Flags::Toggle_showing_goals] == !!(The_mission.game_type & MISSION_TYPE_TRAINING)) {
```
Comment (z64555) [inline code/missionui/missionbrief.cpp:889]: What, "!!" What's that about?

Comment (z64555) [reply code/missionui/missionbrief.cpp:889]: I see now, the `!!` cleans the bit mask its operating on into a proper boolean.

**Pull request merged**